tsdb/head_append.go

```diff
@@ -565,6 +567,7 @@ func (s *memSeries) mmapCurrentHeadChunk(chunkDiskMapper *chunks.ChunkDiskMapper
 	})
 }
 
+// Rollback removes the headAppender's data from the series (TODO is this correct) and writes the series to WAL.
```

Comment:
> removes the headAppender's data from the series

This bit is not correct, it only discards the data from the appender (because we have not added to series yet at this point, it is only the isolation handling that you see below) while WAL logging the series records if any.
```suggestion
// Rollback removes the samples and exemplars from headAppender and writes the series to WAL if any.
```